### ISSUE TITLE
Regarding netmoss score and p-value and their interpretation

### ISSUE THREAD

**Issue by ShailNair** [xiaolw95/NetMoss2#6]
Hi,
My relative abundance table reads like this:

   genus   | control-1 | control-2 | case-1 | case-2 | 
--- | --- | --- | --- | --- |
Colwellia | 0.51 | 0.35 | 43.3 | 56.4 |
Pseudoalteromonas | 0.98 | 0.64 | 23.33 | 24.7 |
Cohaesibacter | 0.07 | 0.05 | 0.07 | 0.1 |
xxxx | xx | xx | xx | xx |

**Netmoss results**

taxon_names| NetMoss_Score | p.val | p.adj | 
--- | --- | --- | --- |
Colwellia| 1 | 0 | 0 | 
Pseudoalteromonas| 0.631 | 2.71E-147 | 6.53E-147 |
Cohaesibacter | 1 | 0.607737938 | 0.61471341 |
....| ... | ... | .. | 

![image](https://user-images.githubusercontent.com/59241611/174463918-cf5c363d-5146-4eb3-bf41-5e59f31981b3.png)

Here the first two genera (Colwellia and Pseudoalteromonas) were highly abundant in CASE samples (relative abundance table). Once I run Netmoss I cannot see their significance (below image). Although Colwellia was enriched in CASE  with a high netmoss score, the p-value is insignificant. On the other hand, Cohaesibacter which did not show much change in its abundance between the two samples (control vs case) shows significantly enriched (with the same netmoss score as that of Collwelia but zero log2FC).

 here, which parameter (netmoss score, p-value (adj.) and Log2FC) is to be considered?

Also why **Cohaesibacter** in the above table have higher netmoss score and p-value than the other two genera (whose relative percentage increased highly in CASE samples)

Note: I run netmoss2 using default parameters

**Comment by xiaolw95** [comment]
NetMoss score is used to evaluate the difference in **network structure** between two groups, which may be not consistent with the difference in abundance.  A higher NetMoss score refers to a greater difference in network structure but not in relative abundance. These two methods assess differential bacteria at different levels and to some extent, are complementary.

**Comment by ShailNair** [comment]
Thank you. What does "p.adj" mean here?. The top 9 genera (top to down) have similar NETMOSS scores. The first two genera were highly enriched in CASE. Then why does only Cohaesibacter is significant here?

**Comment by Damianyangyang** [comment]
> Hi, My relative abundance table reads like this:
> 
> genus	control-1	control-2	case-1	case-2
> Colwellia	0.51	0.35	43.3	56.4
> Pseudoalteromonas	0.98	0.64	23.33	24.7
> Cohaesibacter	0.07	0.05	0.07	0.1
> xxxx	xx	xx	xx	xx
> **Netmoss results**
> 
> taxon_names	NetMoss_Score	p.val	p.adj
> Colwellia	1	0	0
> Pseudoalteromonas	0.631	2.71E-147	6.53E-147
> Cohaesibacter	1	0.607737938	0.61471341
> ....	...	...	..
> ![image](https://user-images.githubusercontent.com/59241611/174463918-cf5c363d-5146-4eb3-bf41-5e59f31981b3.png)
> 
> Here the first two genera (Colwellia and Pseudoalteromonas) were highly abundant in CASE samples (relative abundance table). Once I run Netmoss I cannot see their significance (below image). Although Colwellia was enriched in CASE with a high netmoss score, the p-value is insignificant. On the other hand, Cohaesibacter which did not show much change in its abundance between the two samples (control vs case) shows significantly enriched (with the same netmoss score as that of Collwelia but zero log2FC).
> 
> here, which parameter (netmoss score, p-value (adj.) and Log2FC) is to be considered?
> 
> Also why **Cohaesibacter** in the above table have higher netmoss score and p-value than the other two genera (whose relative percentage increased highly in CASE samples)
> 
> Note: I run netmoss2 using default parameters

Hello，Thank you. In the first table, is the data on diseases and health in a separate directory? When I run NetMoss it always have the error  "Error: id variables not found in data: X".Have you encountered this problem?

**Comment by ShailNair** [comment]
@Damianyangyang Yes. the control group abundance file/s should be in the control **control_dir** while the case group abundance file/s should be in **case_dir**. Also, there should not be any duplicate values in the first column (taxa labels). see [NetMoss2 test data for file setup](https://github.com/xiaolw95/NetMoss2/tree/main/testData).

Here is how I run it:

 **Load packages**
```{r}
library("NetMoss2")

#**Set seed**
sessionInfo()
set.seed(0168)
```

#**Setup directories**
```{r}
setwd('~/Documents/Netmoss_analysis/')   ####set the working directory

#**read directory**
case_dir = paste0(getwd(),"/case")
control_dir = paste0(getwd(),"/control")
net_case_dir = paste0(getwd(),"/net_case_dir")
net_control_dir = paste0(getwd(),"/net_control_dir")

```

#**contruct networks**    ####if files exist, skip
```{r}
netBuild(case_dir = case_dir,
        control_dir = control_dir,
        method = "sparcc")
```
#**calculate NetMoss score**

```{r}
nodes_result = NetMoss(case_dir = case_dir,    
        control_dir = control_dir,    
        net_case_dir = net_case_dir,   
        net_control_dir = net_control_dir) 
```

#**Results**
```{r}
result = nodes_result[[1]]     ####NetMoss score result


write.table(result, "~/Documents//Netmoss_analysis/netmos.results.tsv", sep="\t", quote=F, col.names=NA)

#plot networks
netPlot(nodes_result)    ####image saved in working directory

```

**Comment by Damianyangyang** [comment]
@ShailNair Thank you very much. But I still have some question. When I run it with some author's test data, NetMoss runs successfully. But when I use my own data, I still get this error.
![image](https://user-images.githubusercontent.com/71619162/179436291-e031fb88-6879-4c63-9049-47d6d537c754.png)
And this is my own data. "net_case/control_dir" is generated with netBuild.
[d_study1.txt](https://github.com/xiaolw95/NetMoss2/files/9129080/d_study1.txt)
[h_study1.txt](https://github.com/xiaolw95/NetMoss2/files/9129081/h_study1.txt)
[d_study_net1.txt](https://github.com/xiaolw95/NetMoss2/files/9129083/d_study_net1.txt)
[h_study_net1.txt](https://github.com/xiaolw95/NetMoss2/files/9129084/h_study_net1.txt)

**Comment by ShailNair** [comment]
Did you check your results matrix table. When I ran your data, I got the following result:

![image](https://user-images.githubusercontent.com/59241611/179503419-22771460-d9cc-4f25-ab4b-3c07e7952478.png)

Hence, it is clear why you get id variable not found in data :X.

Based on your data, it appears that you are using filtered data from a few species. It is strongly advised that you use the relative abundance table of all identified taxa. As you are using a relative abundance table, the data is mostly filtered for low variance and normalized for sequencing depth. 


Or you can try with  top 30-100 species. However, this is not advised since it will provide biased results.

**Comment by ShailNair** [comment]
`I always have some errors when using the relative abundance data`

What error you are referring to? you can check the netmoss package document or the [readme file](https://github.com/xiaolw95/NetMoss2) for troubleshooting.  

I could successfully run netmoss2 with your data:
 [d_study1.txt](https://github.com/xiaolw95/NetMoss2/files/9131851/d_study1.txt) [h_study1.txt](https://github.com/xiaolw95/NetMoss2/files/9131852/h_study1.txt) 

While plotting the interaction network, it showed 
`the threshold is too high to find the target taxon in the control networks!Error in if (as.character(edge.control2[mm, 1]) == top.tax2[kk] || as.character(edge.control2[mm,  : `

The default  **e.th**  is **0.4** . by adjusting it to  **0.1** or lower you can visualize the plots. 





`Is it necessary to run “netbuild” with the count of species`
Of course not if you already have the correlation data generated by other algorithms. In both cases, without the correlation data NetMoss cannot run

**Comment by Damianyangyang** [comment]
> 
Thank you for your patience. I got these errors when I used the "netBuild" command to process the above two relative abundance data.
https://github.com/xiaolw95/NetMoss2/files/9131851/d_study1.txt
https://github.com/xiaolw95/NetMoss2/files/9131852/h_study1.txt
`library("NetMoss2")`
`case_dir="/home/bioinfo/sunyy/work/NetMoss/netmoss_CRC/case_dir"`
`control_dir="/home/bioinfo/sunyy/work/NetMoss/netmoss_CRC/control_dir"`
`netBuild(case_dir=case_dir,control_dir=control_dir,method="sparcc")`
![image](https://user-images.githubusercontent.com/71619162/179698277-15c1da99-5b06-467b-a802-83e7c538deae.png)
![image](https://user-images.githubusercontent.com/71619162/179698366-4818f534-dd55-4031-89e7-c12be8e54dd9.png)

**Comment by Damianyangyang** [comment]
@ShailNair Thank you very much.
